content/en/docs/components/trainer/user-guides/local-execution-mode/docker.md

```diff
@@ -0,0 +1,430 @@
+
+**Runtime YAML Example**:
+```yaml
+apiVersion: trainer.kubeflow.org/v1alpha1
+kind: ClusterTrainingRuntime
+metadata:
+  name: torch-custom
+  labels:
+    trainer.kubeflow.org/framework: torch
+spec:
+  mlPolicy:
+    numNodes: 1
+    torch:
+      numProcPerNode: auto
+  template:
+    spec:
+      replicatedJobs:
+        - name: node
+          template:
+            spec:
+              template:
+                spec:
+                  containers:
+                    - name: node
+                      image: myregistry.com/pytorch-custom:latest
+```
+
+## Troubleshooting
+
+### Docker Daemon Not Running
+
+**Error**: `Error while fetching server API version: ('Connection aborted.', ConnectionRefusedError(61, 'Connection refused'))`
+
+**Solution**:
+```bash
+# macOS/Windows: Start Docker Desktop
+# Linux: Start Docker daemon
+sudo systemctl start docker
+
+# Verify Docker is running
+docker ps
+```
+
+### Permission Denied
+
+**Error**: `Got permission denied while trying to connect to the Docker daemon socket`
+
+**Solution** (Linux):
+```bash
+# Add your user to docker group
+sudo usermod -aG docker $USER
+
+# Log out and back in, or run
+newgrp docker
+```
+
+### GPU Not Available in Container
+
+**Error**: `RuntimeError: No CUDA GPUs are available`
+
+**Solution**:
+```bash
+# 1. Verify NVIDIA drivers on host
+nvidia-smi
+
+# 2. Verify NVIDIA Container Toolkit
+docker run --rm --gpus all nvidia/cuda:11.8.0-base-ubuntu22.04 nvidia-smi
+
+# 3. Request GPU in your trainer
+trainer = CustomTrainer(
+    func=train_model,
+    resources_per_node={"gpu": "1"}
+)
+```
+
+### Containers Not Removed
+
+**Problem**: Containers remain after job completion
+
+**Solution**:
+```python
+# Ensure auto_remove is enabled
+backend_config = ContainerBackendConfig(
+    container_runtime="docker",
+    auto_remove=True  # Default
+)
+
+# Or manually clean up
+client.delete_job(job_name)
+
+# Or use Docker CLI
+docker rm -f $(docker ps -aq --filter "label=kubeflow.org/job-name=<job-name>")
+```
+
+### Network Conflicts
+
+**Error**: `network with name <job-name>-net already exists`
+
+**Solution**:
+```bash
+# Remove conflicting network
+docker network rm <job-name>-net
+
+# Or delete the previous job
+# client.delete_job(job_name)
+```
+
+## Docker vs Podman Comparison
+
+| Feature | Docker | Podman |
+|---------|--------|--------|
+| **Architecture** | Client-server (daemon) | Daemonless |
+| **Root Requirement** | Requires root or docker group | Supports rootless mode |
+| **Security** | Daemon runs as root | Better security isolation |
+| **Compatibility** | Wide ecosystem support | Docker-compatible CLI |
+| **Setup Complexity** | Easy (Docker Desktop) | More complex initial setup |
+| **Best For** | General use, macOS/Windows | Security-focused, Linux servers |
+
+To use Podman instead, see the [Container Backend with Podman](./podman.md) guide.
+
+## Next Steps
```

Comment:
Maybe also link to your example here: https://github.com/kubeflow/trainer/blob/master/examples/local/local-container-mnist.ipynb